lnbits/wallets/alby.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+from typing import AsyncGenerator, Optional
+
+import httpx
+from loguru import logger
+
+from lnbits.settings import settings
+
+from .base import (
+    InvoiceResponse,
+    PaymentResponse,
+    PaymentStatus,
+    StatusResponse,
+    Unsupported,
+    Wallet,
+)
+
+
+class AlbyWallet(Wallet):
+    """https://guides.getalby.com/alby-wallet-api/reference/api-reference"""
+
+    def __init__(self):
+        endpoint = settings.alby_api_endpoint
+
+        if not endpoint or not settings.alby_api_key:
+            raise Exception("cannot initialize getalby")
+
+        self.endpoint = endpoint[:-1] if endpoint.endswith("/") else endpoint
+        self.auth = {"Authorization": "Bearer " + settings.alby_api_key}
+        self.client = httpx.AsyncClient(base_url=self.endpoint, headers=self.auth)
+
+    async def cleanup(self):
+        try:
+            await self.client.aclose()
+        except RuntimeError as e:
+            logger.warning(f"Error closing wallet connection: {e}")
+
+    async def status(self) -> StatusResponse:
+        try:
+            r = await self.client.get("/balance", timeout=10)
+        except (httpx.ConnectError, httpx.RequestError):
+            return StatusResponse(f"Unable to connect to '{self.endpoint}'", 0)
+
+        data = r.json()["balance"]
+        if r.is_error:
+            return StatusResponse(data["error"], 0)
+
+        return StatusResponse(None, data)
+
+    async def create_invoice(
+        self,
+        amount: int,
+        memo: Optional[str] = None,
+        description_hash: Optional[bytes] = None,
+        unhashed_description: Optional[bytes] = None,
+        **kwargs,
+    ) -> InvoiceResponse:
+        if description_hash or unhashed_description:
+            raise Unsupported("description_hash")
```

Comment:
you can pass a `description_hash` to create an invoice, is there a reason this is unsupported?

Reply:
give me a few minutes to see if this was an artifact

Reply:
work pulled me away from this, will look at again in about 6-8 hrs

Reply:
see latest commit